src_c/math.c

```diff
@@ -1574,6 +1577,44 @@ vector_str(pgVector *self)
     return Text_FromUTF8(buffer[bufferIdx % 2]);
 }
 
+static PyObject *
+vector_project_onto(pgVector *self, PyObject *other)
+{
+    Py_ssize_t i;
+    pgVector *ret;
+    double other_coords[VECTOR_MAX_SIZE];
+    double factor;
+    double a_dot_b;
+    double b_dot_b;
+
+    if (!PySequence_AsVectorCoords(other, other_coords, self->dim)) {
+        PyErr_SetString(PyExc_TypeError, "Expected Vector as argument 1");
+        return NULL;
+    }
+
+    ret = (pgVector *)pgVector_NEW(self->dim);
+    if (ret == NULL) {
+        return NULL;
+    }
+
+    a_dot_b = _scalar_product(self->coords, other_coords, self->dim);
+    b_dot_b = _scalar_product(other_coords, other_coords, self->dim);
+
+    if (b_dot_b < self->epsilon) {
+        PyErr_SetString(PyExc_ValueError,
+                        "Cannot project onto a vector with zero length");
+        return NULL;
+    }
+
+    factor = a_dot_b / b_dot_b;
+
+    for (i = 0; i < self->dim; ++i)
+        ret->coords[i] = other_coords[i] * factor;
```

Comment:
with the tests in place now maybe this could be optimized for speed. But I will leave it as is for now.